src/hardware/gus.cpp

```diff
@@ -563,22 +569,49 @@ void Voice::WritePanPot(uint8_t pos) noexcept
 // volume scalar value (a floating point fraction between 0.0 and 1.0) is never
 // actually operated on, and is simply looked up from the final index position
 // at the time of sample population.
-void Voice::WriteVolRate(uint16_t val) noexcept
+void Voice::WriteVolRate(const uint16_t val,
+                         const double gus_to_channel_rate_ratio) noexcept
 {
 	vol_ctrl.rate = val;
 	constexpr uint8_t bank_lengths = 63u;
 	const int pos_in_bank = val & bank_lengths;
 	const int decimator = 1 << (3 * (val >> 6));
-	vol_ctrl.inc = ceil_sdivide(pos_in_bank * VOLUME_INC_SCALAR, decimator);
+
+	// Calculate the GUS's native volume increment width
+	const auto gus_vol_inc = ceil_sdivide(pos_in_bank * VOLUME_INC_SCALAR,
+	                                      decimator);
+
+	// Adjust the increment to fit the channel's rate
+	const auto channel_vol_inc = gus_vol_inc * gus_to_channel_rate_ratio;
+	//
+	// Typically the ratio will be less than 1, producing a narrower
+	// increment. By shrinking the increment width, the GUS will produce
+	// more interwave frames (satisfying the channel's greater frame rate
+	// needs), instead of fewer frames for the GUS's lower frame rate.
```

Comment:
This is same as my zero-order-hold resampler, basically, isn't it?

Reply:
GUS's Interwave looks at the time ratio between adjacent samples, and then proportionally interpolates; pretty basic; but a step up from ZoH.

Reply:
Yeah okay, now I remember that the GUS actually does linear interpolation. It has a weird lowpass filter effect when you play samples at lower rates, I remember I was quite taken aback by it, migrating from an Amiga that doesn't do such sample-interpolation at all.

Reply:
Back the day, the closest I got to a GUS was through their magazine ads touting its benefits, and having to pass over the Gravis option in `setup.exe` 😂. 
Very interesting about that low pass effect, and this is the first I've learned of that 👍